### PR TITLE
Player 2194 alt

### PR DESCRIPTION
**Pull request by aeng7**
Muted autoplay workarounds since IMA does not provide the proper hooks to handle muted autoplay

### CODE REVIEW

File: js/ad_manager_template.js [hunk 1]
```diff
@@ -269,6 +269,20 @@ OO.Ads.manager(function(_, $) {
...
+     * initialization. This is done so that the Video Controller can interface with the Ad's Video Plugin
+     * prior to an ad request. A typical use case would be to pass a user click to the ad plugin prior
+     * to ad playback so that the ad can start unmuted for browsers that require user interaction for
+     * unmuted playback.
+     * @method AdManager#createAdVideoElementOnPlayerInit
+     * @public
+     * @returns {string[]} An array of encoding types corresponding to the video elements that the Video Controller
+     *                     should create. Return an empty array, null, or undefined if this is not required.
+     */
+    this.createAdVideoElementOnPlayerInit = function() {
```
Comment (pilievOoyala) [inline js/ad_manager_template.js]: Although it implicitly returns undefined, should we return an empty array?  Up to you.

Comment (aeng7) [reply js/ad_manager_template.js]: I've made a comment saying "Return an empty array, null, or undefined if this is not required." in the jsdoc. I think this is ok. What do you think?

File: js/google_ima.js [hunk 2]
```diff
@@ -2465,7 +2554,10 @@ require("../html5-common/js/utils/utils.js");
      * @public
      * @method TemplateVideoWrapper#unmute
```
Comment (pilievOoyala) [inline js/google_ima.js]: need to document the param

File: js/google_ima.js [hunk 3]
```diff
@@ -2610,6 +2702,14 @@ require("../html5-common/js/utils/utils.js");
       raisePlayhead(this.controller.EVENTS.DURATION_CHANGE, currentTime, duration);
     };
 
+    this.raiseUnmutedPlaybackFailed = function() {
```
Comment (pilievOoyala) [inline js/google_ima.js]: JSDocs?

Comment (pilievOoyala) [reply js/google_ima.js]: Some of these are doc'd and some aren't. Up to you.

File: js/ad_manager_template.js (hunk 1 above)
Comment (pilievOoyala) [reply js/ad_manager_template.js]: That's fine.

**Comment by pilievOoyala** [comment]
LGTM

File: js/ad_manager_template.js (hunk 1 above)
Comment (aeng7) [reply js/ad_manager_template.js]: Done

File: js/google_ima.js (hunk 2 above)
Comment (aeng7) [reply js/google_ima.js]: Done

File: js/google_ima.js (hunk 3 above)
Comment (aeng7) [reply js/google_ima.js]: Done

**Pull request merged**